sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs

```diff
@@ -172,7 +172,7 @@ private async Task<CloudEvent[]> DeserializeCloudEventsInternal(string requestCo
             {
                 cloudEventsInternal.Add(CloudEventInternal.DeserializeCloudEventInternal(requestDocument.RootElement));
             }
-            else // events are wrapped in an array
+            else if (requestDocument.RootElement.ValueKind == JsonValueKind.Array)
```

Comment:
could the ValueKind ever be anything other than Array or Object?

Reply:
In terms of an event that we could deserialize, I don't think so. Should we throw an exception if something else is passed in?

Reply:
I think what you have now is fine. It would just be a no-op if it were some other valueKind. It would be great if we can test for this, though I'm not sure if it is possible.